numpy/_core/_add_newdocs.py

```diff
@@ -4728,6 +4728,81 @@
     its memory, in which case you can traverse ``a.base`` for a memory handler.
     """)
 
+add_newdoc('numpy._core._multiarray_umath', '_array_converter',
+    """
+    _array_converter(*array_likes)
+
+    Helper to convert one or more objects to arrays.  Integrates machinery
+    to deal with the ``result_type`` and ``__array_wrap__``.
+
+    The reason for this is that e.g. ``result_type`` needs to convert to arrays
+    to find the ``dtype``.  But converting to an array before calling
+    ``result_type`` would incorrectly "forget" whether it was a Python int,
+    float, or complex.
+    """)
+
+add_newdoc('numpy._core._multiarray_umath', '_array_converter', ('scalar_input',
```

Comment:
I think `scalar_input` is not used anywhere in the code. Do you have cases in mind where it would be useful? It sort-of feels like it should be, but if there is no use, should it be exposed? (Fine to leave in, just asking.)

Also, right now it is a property. Might it make sense to make it a method (with no arguments), so that one could extend it if needed? (E.g., to ask not just for scalar, but python scalar inputs.)

But perhaps all this just argues for keeping `_array_converter` private for now, so we can see what is best!

Reply:
The linter says this line is one character too long

Reply:
I didn't think we cared, but fair, pushed the fix (inclued it in the previous commit).